src/shared/components/Datepicker/Datepicker.tsx

```diff
@@ -0,0 +1,44 @@
+import React, { forwardRef } from 'react'
+import { parse } from 'date-fns'
+import NumberFormat from 'react-number-format'
+import TextField from '../TextField'
+
+export type DatepickerProps = {
+  required?: boolean
+  error?: boolean
+  disabled?: boolean
+  helperText?: string
+  onChange: (date: Date) => void
```

Comment:
I've used the line I proposed for the story to test it out and noticed that if you type anything and then delete it, the result will be forever the `Invalid Date`. This is a small problem because if a user types something inside that field by mistake and then deletes it, the validation will keep thinking that there's an error there. I suggest we make the onChange have `data: Date | null` and in `handleChange` we call it with `null` if there's no value in the text field. What do you think?